extension-manifest-v2/src/classes/BugDb.js

```diff
@@ -10,246 +10,128 @@
+	);
+	const rawTrackerDB = await response.arrayBuffer();
+	return FiltersEngine.deserialize(new Uint8Array(rawTrackerDB));
+}
 
 /**
  * Class for handling the main Ghostery trackers database
- * which keeps all trackers currently known to Ghostery in
- * a specially structured json format.
- *
- * @extends Updatable
- * @memberOf  BackgroundClasses
  */
-class BugDb extends Updatable {
-	/**
-	 * Select new trackers in the bugs database update.
-	 * @param  {Object} new_apps 	trackers in the new database
-	 * @param  {Object} old_apps 	trackers in the original database
-	 * @return {Object}          	list of all new trackers
-	 */
-	static updateNewAppIds(new_apps, old_apps) {
-		log('updating newAppIds...');
-
-		const new_app_ids = difference(
-			keys(new_apps),
-			keys(old_apps)
-		).map(Number);
-
-		conf.new_app_ids = new_app_ids;
-
-		return new_app_ids;
+export class BugDb {
+	constructor() {
+		this.db = {
+			categories: [],
+			apps: {},
+			bugs: {},
+			version: 0,
+		};
+		/* keep conf as property for sake of testing */
+		this.conf = conf;
+		this.engine = null;
 	}
 
-	/**
-	 * Build trackers into category arrays, for use on Global Blocking UI list. This is
-	 * called once on initial startup.
-	 *
-	 * @private
-	 *
-	 * @param  {Object} db      bugs database object
-	 * @return {array}  		array of categories
-	 */
-	static _buildCategories(db) {
-		const selectedApps = conf.selected_app_ids || {};
-		let appId;
-		let category;
-		let blocked;
-
-		const categoryArray = [];
-		const categories = {};
-
-		const appIds = Object.keys(db.apps);
-		for (let i = 0; i < appIds.length; i++) {
-			appId = appIds[i];
-			category = db.apps[appId].cat;
-			if (t(`category_${category}`) === `category_${category}`) {
-				category = 'uncategorized';
-			}
-			blocked = selectedApps.hasOwnProperty(appId);
-
-			// Because we have two trackers in the DB with the same name
-			if ((categories[category] && categories[category].trackers[db.apps[appId].name])) {
-				continue; // eslint-disable-line no-continue
-			}
-
-			if (categories.hasOwnProperty(category)) {
-				categories[category].num_total++;
-				if (blocked) {
-					categories[category].num_blocked++;
-				}
-			} else {
-				categories[category] = {
-					id: category,
-					name: t(`category_${category}`),
-					description: t(`category_${category}_desc`),
-					img_name: (category === 'advertising') ? 'adv' : // Because AdBlock blocks images with 'advertising' in the name.
-						(category === 'social_media') ? 'smed' : category, // Because AdBlock blocks images with 'social' in the name.
-					num_total: 1,
-					num_blocked: (blocked) ? 1 : 0,
-					trackers: []
-				};
-			}
-			categories[category].trackers.push({
-				id: appId,
-				name: db.apps[appId].name,
-				description: '',
-				blocked,
-				shouldShow: true,
-				catId: category,
-				trackerID: db.apps[appId].trackerID,
-			});
+	async init(just_upgraded) {
```

Comment:
Is it possible that this function gets called multiple times in parallel? If so, the engine might get loaded multiple times. (Which would not be a correctness problem, but would be wasteful.)

Reply:
the only place it is being called is during background startup